url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/fri/patterns/interpreter/parsergenerator/syntax/builder/SyntaxBuilderParserTables.java

```diff
@@ -0,0 +1,720 @@
+		loadParseAction_25();
+		loadParseAction_26();
+		loadParseAction_27();
+		loadParseAction_28();
+		loadParseAction_29();
+		loadParseAction_30();
+		loadParseAction_31();
+		loadParseAction_32();
+		loadParseAction_33();
+		loadParseAction_34();
+	}
+	private void loadParseAction_0()	{
+		Hashtable p = new Hashtable(1, 1);
+		parseTable.add(p);
+		p.put("`identifier`", new Integer(-2));
+	}
+	private void loadParseAction_1()	{
+		Hashtable p = new Hashtable(1, 1);
+		parseTable.add(p);
+		p.put("\"::=\"", new Integer(-2));
+	}
+	private void loadParseAction_2()	{
+		Hashtable p = new Hashtable(2, 1);
+		parseTable.add(p);
+		p.put("`identifier`", new Integer(2));
+		p.put("\"EoI\"", new Integer(2));
+	}
+	private void loadParseAction_3()	{
+		Hashtable p = new Hashtable(2, 1);
+		parseTable.add(p);
+		p.put("`identifier`", new Integer(-2));
+		p.put("\"EoI\"", new Integer(0));
+	}
+	private void loadParseAction_4()	{
+		Hashtable p = new Hashtable(7, 1);
+		parseTable.add(p);
+		p.put("'|'", new Integer(22));
+		p.put("'('", new Integer(-2));
+		p.put("';'", new Integer(22));
+		p.put("`bnf_chardef`", new Integer(-2));
+		p.put("`ruleref`", new Integer(-2));
+		p.put("`identifier`", new Integer(-2));
+		p.put("`stringdef`", new Integer(-2));
+	}
+	private void loadParseAction_5()	{
+		Hashtable p = new Hashtable(2, 1);
+		parseTable.add(p);
+		p.put("`identifier`", new Integer(1));
+		p.put("\"EoI\"", new Integer(1));
+	}
+	private void loadParseAction_6()	{
+		Hashtable p = new Hashtable(2, 1);
+		parseTable.add(p);
+		p.put("';'", new Integer(-2));
+		p.put("'|'", new Integer(-2));
+	}
+	private void loadParseAction_7()	{
+		Hashtable p = new Hashtable(11, 1);
+		parseTable.add(p);
+		p.put("`identifier`", new Integer(7));
+		p.put("'('", new Integer(7));
+		p.put("`stringdef`", new Integer(7));
+		p.put("`quantifier`", new Integer(7));
+		p.put("`ruleref`", new Integer(7));
+		p.put("'-'", new Integer(7));
+		p.put("')'", new Integer(7));
+		p.put("';'", new Integer(7));
+		p.put("`bnf_chardef`", new Integer(7));
+		p.put("'|'", new Integer(7));
+		p.put("\"..\"", new Integer(-2));
+	}
+	private void loadParseAction_8()	{
+		Hashtable p = new Hashtable(9, 1);
+		parseTable.add(p);
+		p.put("`ruleref`", new Integer(14));
+		p.put("';'", new Integer(14));
+		p.put("'('", new Integer(14));
+		p.put("'|'", new Integer(14));
+		p.put("`quantifier`", new Integer(14));
+		p.put("')'", new Integer(14));
+		p.put("`identifier`", new Integer(14));
+		p.put("`stringdef`", new Integer(14));
+		p.put("`bnf_chardef`", new Integer(14));
+	}
+	private void loadParseAction_9()	{
+		Hashtable p = new Hashtable(8, 1);
+		parseTable.add(p);
+		p.put("`bnf_chardef`", new Integer(-2));
+		p.put("`ruleref`", new Integer(-2));
+		p.put("'|'", new Integer(21));
+		p.put("')'", new Integer(21));
+		p.put("'('", new Integer(-2));
+		p.put("';'", new Integer(21));
+		p.put("`stringdef`", new Integer(-2));
+		p.put("`identifier`", new Integer(-2));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 418:427, 562:571.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.